capstone-backend/src/main/java/com/google/sps/servlets/BookListServlet.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.BookList;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.base.Joiner;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that stores, updates and gets BookList's data from Frontend . */
+@WebServlet("/api/booklist")
```

Comment:
If you've pulled changes to the Utility class into this PR, it could be nice to use those methods here rather than re-creating them yourself. This refactoring can also be done later.

Reply:
ok